src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs

```diff
@@ -366,10 +361,9 @@ private TestParameters WithScriptOptions(TestParameters parameters)
                 workspace.Options = workspace.Options.WithChangedOption(
                     RemoteFeatureOptions.DiagnosticsEnabled, false);
 
-                var actions = await GetCodeActionsAsync(workspace, parameters);
+                var (actions, actionIndex) = await GetCodeActionsAsync(workspace, parameters, index);
                 await TestActionsAsync(
-                    workspace, expected, index,
-                    actions,
+                    workspace, expected, actionIndex, actions,
```

Comment:
note: i'm not a huge fan of this pattern i'm using here.  Effectively, the test says "the user wants to invoke the action at index 'N'".  We then go and pass that along to GetCodeActions which will find the action at that index.  However, if fix-all is requested, a *new* code action for the fix-all is synthesized and it is *that* action that is returned.  Since htat is the only action returned, we clearly cannot index into the returned list with the original "index", so the callee returns the index that should be used.

A much cleaner way to do this would be to just return the action to invoke.  However, we have a bunch of code that examines all the actions (for thigns like checking spans/titles/etc.).  

However, i think i can clean this up by returning the list of actions and the specific action to invoke.  i will try to make that change next.